contrib/opencensus-ext-azure/opencensus/ext/azure/common/utils.py

```diff
@@ -65,3 +66,23 @@ def timestamp_to_iso_str(timestamp):
 
 def url_to_dependency_name(url):
     return urlparse(url).netloc
+
+
+# Validate UUID format
+# Specs taken from https://tools.ietf.org/html/rfc4122
+pattern = re.compile('[0-9a-f]{8}-'
+                     '[0-9a-f]{4}-'
+                     '[1-5][0-9a-f]{3}-'
+                     '[89ab][0-9a-f]{3}-'
+                     '[0-9a-f]{12}')
+
+
+def validate_key(instrumentation_key):
```

Comment:
I assume "prefixed" ikeys, e.g. `AIF-1234abcd-5678-4efa-8abc-1234567890ab` would be rejected by this. Is this strict checking something we want to expose a way to turn off?

Reply:
We've decided that checking for GUID is sufficient enough for validation. As for "prefixed" ikeys, the specs that were given for prefixed ikeys do not seem to apply for AI customers, so we can safely assume that all ikeys should be GUIDs.